tests/pvc_utils.go

```diff
@@ -0,0 +1,74 @@
+package tests
+
+import (
+	"time"
+
+	. "github.com/onsi/ginkgo"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	defaultTimeout = 30*time.Second
+)
+
+// Creates a standard PVC in the passed in namespace.
+func CreateDefaultPVC(client *kubernetes.Clientset, namespace string, name string, size string) *k8sv1.PersistentVolumeClaim {
+	return CreatePVCFromDefinition(client, namespace, NewPVCDefinition(name, size, nil))
+}
+
+// Creates a PVC in the passed in namespace from the passed in PersistentVolumeClaim definition.
+func CreatePVCFromDefinition(client *kubernetes.Clientset, namespace string, def *k8sv1.PersistentVolumeClaim) *k8sv1.PersistentVolumeClaim {
+	var pvc *k8sv1.PersistentVolumeClaim
+	var err error
+	if wait.PollImmediate(2*time.Second, defaultTimeout, func() (bool, error) {
+		pvc, err = client.CoreV1().PersistentVolumeClaims(namespace).Create(def)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}) != nil {
+		Fail("Unable to create PVC: " + def.GetName())
+	}
+	return pvc
+}
+
+// Delete the passed in PVC
+func DeletePVC(client *kubernetes.Clientset, namespace string, pvc *k8sv1.PersistentVolumeClaim) {
+	if wait.PollImmediate(2*time.Second, defaultTimeout, func() (bool, error) {
+		err := client.CoreV1().PersistentVolumeClaims(namespace).Delete(pvc.GetName(), nil)
+		if err != nil {
+			return false, nil
+		}
+		return true, nil
+	}) != nil {
+		Fail("Unable to delete PVC: " + pvc.GetName())
+	}
+}
+
+// Creates a PVC definition using the passed in name and requested size.
+// You can use the following annotation keys to request an import or clone. The values are defined in the controller package
+// AnnEndpoint
+// AnnSecret
+// AnnCloneRequest
+func NewPVCDefinition(name string, size string, annotations map[string]string) *k8sv1.PersistentVolumeClaim {
```

Comment:
maybe namespace too?

Reply:
Nevermind, I see ns is in the Create() call.

Reply:
Yeah, its a choice of adding it here or in the create call. Either would work.